api/Map.js

```diff
@@ -0,0 +1,209 @@
+import EPSG_21781 from '@geoblocks/sources/EPSG21781.js';
+
+
+class Map {
+
+  /**
+   * @param {Object} options
+   * @property {string} div
+   * @property {ol.Coordinate} center
+   * @property {number} [zoom=10]
+   * TODO: more options
+   */
+  constructor(options) {
+    /**
+     * @private
+     * @type {View}
+     */
+    this.view_ = new View({
+      projection: getProjection(constants.projection || 'EPSG:21781'),
+      resolutions: constants.resolutions,
+      zoom: options.zoom,
+      center: options.center
+    });
+
+    /**
+     * @private
+     * @type {OLMap}
+     */
+    this.map_ = new OLMap({
+      controls: [],
+      target: options.div,
+      view: this.view_
+    });
+
+    // FIXME: temporary
+    const background = new TileLayer({
+      source: new SwisstopoSource({
+        layer: 'ch.swisstopo.pixelkarte-farbe',
+        format: 'image/jpeg',
+        timestamp: 'current',
+        projection: EPSG_21781,
+        crossOrigin: 'anonymous'
+      })
+    });
+    this.map_.addLayer(background);
+
+    /**
+     * @private
+     * @type {VectorSource}
+     */
+    this.vectorSource_ = new VectorSource();
+
+    /**
+     * @private
+     * @type {VectorLayer}
+     */
+    this.vectorLayer_ = new VectorLayer({
+      source: this.vectorSource_
+    });
+
+    this.map_.addLayer(this.vectorLayer_);
+  }
+
+  /**
+   * @param {ol.Coordinate} center
+   * @param {number} zoom
+   */
+  recenter(center, zoom) {
+    this.view_.setCenter(center);
+    this.view_.setZoom(zoom);
+  }
+
+  /**
+   * @param {Object} options
+   * @property {ol.Coordinate} position
+   * @property {string} [icon]
+   * @property {ol.Size} [size]
+   */
+  addMarker(options = {}) {
+    const marker = new Feature({
+      geometry: new Point(options.position ? options.position : this.view_.getCenter())
+    });
+    if (options.icon) {
+      // FIXME: use size?
+      marker.setStyle(new Style({
+        image: new Icon({
+          src: options.icon
+        })
+      }));
+    }
+    this.vectorSource_.addFeature(marker);
+  }
+
+  /**
+   * @param {string} layer
+   * @param {Array.<string>} layer
+   * @param {boolean} [highlight=false]
+   */
+  recenterOnObjects(layer, ids, highlight = false) {
+
+  }
+
+  /**
+   * @param {string} type
+   * @param {string} name
+   * @param {string} url
+   * @param {Object} [options]
+   * @property {Array.<string>} [attr=['title', 'description']]
+   * @property {function()} [success]
+   * @property {function()} [error]
+   */
+  addCustomLayer(type, name, url, options = {}) {
+    if (type === 'text') {
+      fetch(url)
+        .then((response) => response.text())
+        .then((text) => {
+          const attr = options.attr || ['title', 'description'];
+          const lines = text.split(/\r\n|\r|\n/);
+          const columns = lines.shift().split('\t');
+          for (const line of lines) {
+            if (line) {
+              const values = zip(columns, line.split('\t'));
```

Comment:
(It is cross compatible ? It is always` \t` ?)

Reply:
yes, it's always `\t`. It was like this in openlayers 2:
https://github.com/openlayers/ol2/blob/master/lib/OpenLayers/Format/Text.js#L91

Reply:
Ok, I was just suspicious